### PR TITLE
[mono][debugger] Support new JMC

### PR DESCRIPTION
**Pull request by thaystg**
Not sure if this is the best approach yet, but for now is working.

### CODE REVIEW

File: src/mono/mono/component/debugger-agent.c [hunk 1]
```diff
@@ -7424,7 +7424,111 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
...
+	*size = (StepSize)decode_int (p, &p, end);
+	*depth = (StepDepth)decode_int (p, &p, end);
+	int module_len = decode_int (p, &p, end);
+	MonoMethod *paused_method = NULL;
+	modifier->data.jmc = (JMC_Modifier *)g_malloc0 (sizeof (JMC_Modifier));
+	if (*depth == STEP_DEPTH_OVER) {
+		g_assert (*step_thread_id);
+		err = get_object (*step_thread_id, (MonoObject**)&step_thread);
+		if (err != ERR_NONE) {
+			g_free (req);
```
Comment (radical) [inline src/mono/mono/component/debugger-agent.c:7466]: curious - why is this freed in this code path, but not for the others in this function?

**Comment by stephentoub** [comment]
@thaystg, are you still working on this? The last update was over a year ago. Should it be closed?

**Comment by thaystg** [comment]
We can close.

**Pull request closed without merging**